src/components/MyRecipes.vue

```diff
@@ -1,291 +1,566 @@
+          // Make the API call to fetch the ingredient thumbnails
+          const response = await this.fetch_ingredient_thumbnail(ingredient)
+          // Extract the thumbnail URLs from the API response
+          const imageResults = response;
+          // Update the corresponding ingredient object with the image results
+          this.$set(this.ingredients[index], 'imageResults', imageResults);
+          this.$set(this.ingredients[index], 'showImageResults', imageResults.length > 0);
+        } catch (error) {
+          console.log(error)
+        } finally {
+          this.$set(this.ingredients[index], 'loading', false);
         }
+      }
     },
-    components: {
-        UploadImage,
-        VueEditor,
-        FlagPicker
+
+    fetch_ingredient_thumbnail(ingredient) {
+      const client = createClient(process.env.PEXEL_API_KEY);
+      const query = ingredient;
+      return new Promise((resolve, reject) => {
+        client.photos.search({ query, per_page: 3 })
+          .then(photos => {
+            resolve(photos.photos)
+          })
+          .catch(error => {
+            console.log(error)
+            reject(error)
+          })
+      })
+    },
+
+    selectThumbnail(index, thumbnailUrl) {
+      this.ingredients[index].thumbnail = thumbnailUrl;
     },
-    methods: {
-        toggleEditor(action) {
-            if (action === 'hide') {
-                $("#recipe-editor").removeClass("show")
-                $("#recipe-editor").addClass("hide")
-            }
-
-            if (action === 'show') {
-                $("#recipe-editor").addClass("show")
-                $("#recipe-editor").removeClass("hide")
-            }
-            this.inEditMode = !this.inEditMode
-        },
-        addField(value, field) {
-            field.push({ value: { name: "", unit: "", thumbnail: "", link: "" } })
-        },
-        removeField(index, field) {
-            field.splice(index, 1);
+
+    dropdownValidation: function () {
+      let isValid = true;
+      if (!this.nationality) {
+        this.nationalityError = "The nationality field is required"
+        isValid = false;
+      }
+      if (!this.cookbook_id) {
+        this.cookbookError = "The cookbook field is required";
+        isValid = false
+      }
+      return isValid
+    },
+
+    async submitButton() {
+      const validFile = this.dropdownValidation()
+      this.submitLoading = true
+      const dataSend =
+      {
+        title: this.title,
+        nationality: this.nationality,
+        ingredients: this.ingredients,
+        keywords: this.keywords,
+        draft: "false",
+        recipeDescription: this.recipeDescription,
+        imagePath: this.imagePath,
+        cookbook_id: this.cookbook_id,
+        tags: [],
+        summary: this.summary,
+      }
+
+      if (validFile === true) {
+        const postResponse = await this.$store.dispatch('post_recipe', dataSend)
+        const result = await postResponse
```

Comment:
Let's not use try catch either, we already talked about that in another issue you worked on. Let's use the handlers that axios provides to handle success and failure